src/Microsoft.DocAsCode.Dfm/MarkdownValidators/MarkdownValidatorBuilder.cs

```diff
@@ -15,75 +16,202 @@ namespace Microsoft.DocAsCode.Dfm.MarkdownValidators
+            foreach (var item in _validators)
             {
-                foreach (IMarkdownTokenValidatorProvider vp in CompositionHost.GetExports(typeof(IMarkdownTokenValidatorProvider), contract))
+                if (IsDisabledBySetting(item) ?? item.Rule.Disable)
                 {
-                    list.AddRange(vp.GetValidators());
+                    enabledContractName.Remove(item.Rule.RuleName);
                 }
+                else
+                {
+                    enabledContractName.Add(item.Rule.RuleName);
+                }
+            }
+            foreach (var pair in _globalValidators)
+            {
+                if (pair.Value.Disable)
+                {
+                    enabledContractName.Remove(pair.Value.RuleName);
+                }
+                else
+                {
+                    enabledContractName.Add(pair.Value.RuleName);
+                }
+            }
+            return from name in enabledContractName
+                   from IMarkdownTokenValidatorProvider vp in CompositionHost.GetExports(typeof(IMarkdownTokenValidatorProvider), name)
+                   from v in vp.GetValidators()
+                   select v;
+        }
+
+        private IEnumerable<MarkdownTagValidationRule> GetEnabledTagRules()
+        {
+            foreach (var item in _tagValidators)
+            {
+                if (IsDisabledBySetting(item) ?? item.Rule.Disable)
+                {
+                    continue;
+                }
+                yield return item.Rule;
             }
-            var context = new MarkdownRewriterContext(CompositionHost, TagValidators);
-            list.Add(MarkdownTokenValidatorFactory.FromLambda<MarkdownTagInlineToken>(context.Validate));
-            return MarkdownTokenRewriterFactory.FromLambda(
-                (IMarkdownRewriteEngine engine, IMarkdownToken token) =>
+        }
+
+        private bool? IsDisabledBySetting<T>(RuleWithId<T> item)
+        {
+            bool? categoryDisable = null;
+            bool? idDisable = null;
+            if (item.Category != null)
+            {
+                foreach (var setting in _settings)
                 {
-                    using (new LoggerPhaseScope(MarkdownValidatePhaseName))
+                    if (setting.Category == item.Category)
                     {
-                        foreach (var item in list)
+                        if (setting.Id == null)
+                        {
+                            categoryDisable = setting.Disable;
+                        }
+                        else if (setting.Id == item.Id)
                         {
-                            item.Validate(token);
+                            idDisable = setting.Disable;
                         }
                     }
-                    return null;
-                });
+                }
+            }
+            return categoryDisable ?? idDisable;
```

Comment:
bug, wrong order...

---
In reply to: [73126354](https://github.com/dotnet/docfx/pull/545#discussion_r73126354) [](ancestors = 73126354)